### PR TITLE
7044 address TodoOrDie, remove deprecated overrides

### PR DESCRIPTION
**Pull request by ttoomey**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch

## Description
[GH Issue](https://github.com/open-path/Green-River/issues/7044)

* remove deprecated override fields from inventory, project, and projectCoC
  * propagate changes to analytics views 
* remove cookie rotator from rails 7 upgrade
* remove helper code for unused index removal
* remove unused es_nbn_pre_2024? and tracking_method_to_use

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [x] I have performed a self-review of my code
- [ ] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] I have updated the documentation (or not applicable)
- [ ] I have added spec tests (or not applicable)
- [ ] I have provided testing instructions in this PR or the related issue (or not applicable)

### CODE REVIEW

File: app/models/report_generators/pit/fy2018/base.rb [hunk 1]
```diff
@@ -845,7 +845,7 @@ def chronic_scope
 
     def sh_cols
       @sh_cols ||= {
-        project_type: act_as_project_overlay,
+        project_type: she_s[:project_type].as('client_id'),
```
Comment (eanders) [inline app/models/report_generators/pit/fy2018/base.rb]:
```suggestion
        project_type: she_s[:project_type].as('project_type'),
```

**Review by eanders — APPROVED** [review]
Thanks @ttoomey !  Only one comment, I think the sql statement in the PIT base.rb file needs a tweak.

File: app/models/report_generators/pit/fy2018/base.rb (hunk 1 above)
Comment (ttoomey) [reply app/models/report_generators/pit/fy2018/base.rb]: Thanks for spotting! I had the wrong table alias too 🤦

**Comment by ttoomey** [comment]
@eanders - thanks for the review! Should this go into 144?

**Comment by eanders** [comment]
143 I think, or we'll need to push the dates back on 143.

File: db/warehouse/migrate/20241203154146_remove_deprecated_overrides.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      ['Project', 'hmis_participating_project_override', :integer],
+      ['Project', 'target_population_override', :integer],
+      ['Project', 'tracking_method_override', :integer], # extra
+      ['ProjectCoC', 'hud_coc_code', :string],
+      ['ProjectCoC', 'geography_type_override', :integer],
+      ['ProjectCoC', 'geocode_override', :string],
+      ['ProjectCoC', 'zip_override', :string],
+    ].each do |table, field, _type|
+      # remove_column table, field, type, if_exists: true
+      rename_column table, field, "deleted_#{field}" if column_exists?(table, field)
```
Comment (eanders) [inline db/warehouse/migrate/20241203154146_remove_deprecated_overrides.rb]: I think the `if column_exists...` should only happen on up, or should check a different name on `down`.

Comment (ttoomey) [reply db/warehouse/migrate/20241203154146_remove_deprecated_overrides.rb]: good catch, thanks. Should be fixed

**Pull request merged**